File: docs/snippets/python/cache/disk_cache.py
```python
from pathlib import Path
from kreuzberg import Kreuzberg, ExtractionConfig, CacheConfig

cache_dir = Path.home() / ".cache" / "kreuzberg"
cache_dir.mkdir(parents=True, exist_ok=True)

config = ExtractionConfig(
    use_cache=True,
    cache_config=CacheConfig(
        cache_path=str(cache_dir),
        max_cache_size=500 * 1024 * 1024,
        cache_ttl_seconds=7 * 86400,
        enable_compression=True,
    ),
)

kreuzberg = Kreuzberg(config)

print("First extraction (will be cached)...")
result1 = kreuzberg.extract_file("document.pdf")
print(f"  - Content length: {len(result1.content)}")
print(f"  - Cached: {result1.metadata.get('was_cached', False)}")

print("\nSecond extraction (from cache)...")
result2 = kreuzberg.extract_file("document.pdf")
print(f"  - Content length: {len(result2.content)}")
print(f"  - Cached: {result2.metadata.get('was_cached', False)}")

print(f"\nResults are identical: {result1.content == result2.content}")

cache_stats = kreuzberg.get_cache_stats()
print(f"\nCache Statistics:")
print(f"  - Total entries: {cache_stats.get('total_entries', 0)}")
print(f"  - Cache size: {cache_stats.get('cache_size_bytes', 0) / 1024 / 1024:.1f} MB")
print(f"  - Hit rate: {cache_stats.get('hit_rate', 0):.1%}")
```

File: docs/snippets/go/cache/disk_cache.go
```go
package main

import (
	"fmt"
	"os"
	"path/filepath"

	"kreuzberg"
)

func main() {
	cacheDir := filepath.Join(os.Getenv("HOME"), ".cache", "kreuzberg")
	os.MkdirAll(cacheDir, 0755)

	config := &kreuzberg.ExtractionConfig{
		UseCache: true,
		CacheConfig: &kreuzberg.CacheConfig{
			CachePath:       cacheDir,
			MaxCacheSize:    500 * 1024 * 1024,
			CacheTTLSeconds: 7 * 86400,
			EnableCompression: true,
		},
	}

	client, err := kreuzberg.New(config)
	if err != nil {
		panic(err)
	}
	defer client.Close()

	fmt.Println("First extraction (will be cached)...")
	result1, err := client.ExtractFile("document.pdf")
	if err != nil {
		panic(err)
	}
	fmt.Printf("  - Content length: %d\n", len(result1.Content))
	fmt.Printf("  - Cached: %v\n", result1.Metadata["was_cached"])

	fmt.Println("\nSecond extraction (from cache)...")
	result2, err := client.ExtractFile("document.pdf")
	if err != nil {
		panic(err)
	}
	fmt.Printf("  - Content length: %d\n", len(result2.Content))
	fmt.Printf("  - Cached: %v\n", result2.Metadata["was_cached"])

	fmt.Printf("\nResults are identical: %v\n", result1.Content == result2.Content)

	stats, err := client.GetCacheStats()
	if err != nil {
		panic(err)
	}
	fmt.Println("\nCache Statistics:")
	fmt.Printf("  - Total entries: %d\n", stats.TotalEntries)
	fmt.Printf("  - Cache size: %.1f MB\n", float64(stats.CacheSizeBytes)/1024/1024)
	fmt.Printf("  - Hit rate: %.1f%%\n", stats.HitRate*100)
}
```
